### PR TITLE
Mutations article edits for 3.4

### PR DESCRIPTION
**Pull request by StephenBarlow**

### CODE REVIEW

**Comment by rektide** [comment]
This will be a huge help to our team, thank you.

**Comment by benjamn** [comment]
@StephenBarlow As I'm working on this article, I noticed we're using React refs in a slightly strange way. Do you think this kind of change is a good idea, or is it more confusing for folks who don't know about `useRef`?
```diff
diff --git a/docs/source/data/mutations.mdx b/docs/source/data/mutations.mdx
index 9ed27e9b4..25a3e227a 100644
--- a/docs/source/data/mutations.mdx
+++ b/docs/source/data/mutations.mdx
@@ -73,7 +73,7 @@ Next, we'll create a component named `AddTodo` that represents the submission fo
 
 ```jsx{3,13}:title=add-todo.jsx
 function AddTodo() {
-  let input;
+  const inputRef = useRef();
   const [addTodo, { data, loading, error }] = useMutation(ADD_TODO);
 
   if (loading) return 'Submitting...';
@@ -84,15 +84,12 @@ function AddTodo() {
       <form
         onSubmit={e => {
           e.preventDefault();
+          const input = inputRef.current;
           addTodo({ variables: { text: input.value } });
           input.value = '';
         }}
       >
-        <input
-          ref={node => {
-            input = node;
-          }}
-        />
+        <input ref={inputRef} />
         <button type="submit">Add Todo</button>
       </form>
     </div>

```

**Comment by benjamn** [comment]
@StephenBarlow I enabled the Netlify previews for any PR targeting `release-3.4`, so that's working now, but I'm stuck on getting [`<MutationOptions/>`](https://deploy-preview-8265--apollo-client-docs.netlify.app/docs/react/data/mutations/#options) to rerender with the new (incomplete) `onQueryUpdated` section. Is there anything special I need to do to regenerate MDX components like `<MutationOptions/>`?

File: docs/source/data/mutations.mdx [hunk 1]
```diff
@@ -266,54 +262,44 @@ function AddTodo() {
...
+### Refetching after `update`
 
-Both of these guesses could be slightly wrong in some cases. The `optimisticResponse` eventually gets superseded by the real mutation response, but what can you do to check the guesswork of your `update` function?
+An `update` function attempts to replicate a mutation's back-end modifications in your client's local cache. These cache modifications are broadcast to all affected active queries, which updates your UI automatically. If the `update` function does this correctly, your users see the latest data immediately, without needing to await another network round trip.
 
-> Thankfully, although GraphQL mutations can have arbitrary effects on the server, a GraphQL client only needs to worry about mutation effects that can be later observed by refetching GraphQL queries.
+However, an `update` function might get this replication _wrong_ by setting a cached value incorrectly. You can "double check" your `update` function's modifications by refetching affected active queries on your GraphQL server. To specify _which_ active queries you want to refetch, provide the `onQueryUpdated` option to your mutate function:
```
Comment (benjamn) [inline docs/source/data/mutations.mdx:283]: @StephenBarlow This last sentence seems a bit misleading: `onQueryUpdated` by itself is never the cause of a query being refetched, but merely a way of intercepting/filtering queries that were affected by the mutation's `update` function or included in the `refetchQueries` array. I don't think this confusion spills over into other parts of this documentation, so I believe this sentence can be reworded in place.

If that's not enough information to figure out the right wording, I'm happy (as always) to answer questions, but you might also want to read through the [Refetch recipes](https://deploy-preview-8265--apollo-client-docs.netlify.app/docs/react/data/refetching/#refetch-recipes) section that I just added, to get a feeling for the relationship between the different `client.refetchQueries` options.

**Review by benjamn — APPROVED** [review]
Content looks complete to me, though I'm sure there are some formatting tweaks (especially for the tables I added) that we can continue iterating on (either here or in follow-up PRs). Thanks @StephenBarlow!

**Comment by benjamn** [comment]
Interesting (re: [that Netlify build failure](https://app.netlify.com/sites/apollo-client-docs/deploys/60f9c9f09e186b000846394e))… broken links don't show up for the commit that breaks them, but in the following commit (I think). I'll push a fix momentarily.

**Pull request merged**